### PR TITLE
Subsample

### PR DESCRIPTION
**Pull request by mortonjt**
This addresses #1526 

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/COPYING.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,9 @@
 ## Version 0.5.1-dev (changes since 0.5.1 go here)
 
 ### Features
-* Added ``skbio.io.format.gff3`` for reading and writing GFF3 files for ``DNA``, ``Sequence``, and ``IntervalMetadata`` classes. ([#1450](https://github.com/biocore/scikit-bio/pull/1450))
+* Removing ValueError check in `skbio.stats._subsample.subsample_counts` when `replace=True` [#1527](https://github.com/biocore/scikit-bio/pull/1527)
+
+* ``skbio.io.format.gff3`` for reading and writing GFF3 files for ``DNA``, ``Sequence``, and ``IntervalMetadata`` classes. ([#1450](https://github.com/biocore/scikit-bio/pull/1450))
```
Comment (RNAer) [inline CHANGELOG.md]: why adding the gff3 stuff here?

**Comment by mortonjt** [comment]
It's not. This is being added above

**Comment by RNAer** [comment]
ah, sorry for my mistake. could you check why travis failed?

File: skbio/stats/_subsample.py [hunk 2]
```diff
@@ -231,7 +231,7 @@ def subsample_counts(counts, n, replace=False):
         raise ValueError("Only 1-D vectors are supported.")
 
     counts_sum = counts.sum()
-    if n > counts_sum:
+    if n > counts_sum and not replace:
```
Comment (jairideout) [inline skbio/stats/_subsample.py]: Thanks for the fix! Can you update the `ValueError` below to indicate the issue applies to subsampling without replacement? Also can you update line 176 in the docstring above that describes when this `ValueError` is raised?

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,9 @@
 ## Version 0.5.1-dev (changes since 0.5.1 go here)
 
 ### Features
-* Added ``skbio.io.format.gff3`` for reading and writing GFF3 files for ``DNA``, ``Sequence``, and ``IntervalMetadata`` classes. ([#1450](https://github.com/biocore/scikit-bio/pull/1450))
+* Removing ValueError check in `skbio.stats._subsample.subsample_counts` when `replace=True` [#1527](https://github.com/biocore/scikit-bio/pull/1527)
```
Comment (jairideout) [inline CHANGELOG.md]: *... when `replace=True` and `n` is greater than the number of items in `counts`.*

File: skbio/stats/_subsample.py (hunk 2 above)
Comment (mortonjt) [inline skbio/stats/_subsample.py]: What do you think about the new error message?

Comment (RNAer) [inline skbio/stats/_subsample.py]: it looks like a pandas problem. it should be addressed separately. the previous merged PRs also failed to pass on those 2 python versions.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,9 @@
 ## Version 0.5.1-dev (changes since 0.5.1 go here)
 
 ### Features
-* Added ``skbio.io.format.gff3`` for reading and writing GFF3 files for ``DNA``, ``Sequence``, and ``IntervalMetadata`` classes. ([#1450](https://github.com/biocore/scikit-bio/pull/1450))
+* Removing ValueError check in `skbio.stats._subsample.subsample_counts` when `replace=True` and `n` is greater than the number of items in counts.  [#1527](https://github.com/biocore/scikit-bio/pull/1527)
+
+* ``skbio.io.format.gff3`` for reading and writing GFF3 files for ``DNA``, ``Sequence``, and ``IntervalMetadata`` classes. ([#1450](https://github.com/biocore/scikit-bio/pull/1450))
```
Comment (jairideout) [inline CHANGELOG.md]: Looks like you dropped the start of this sentence ("Added")

File: skbio/stats/_subsample.py (hunk 2 above)
Comment (jairideout) [inline skbio/stats/_subsample.py]: Looks great @mortonjt!

File: CHANGELOG.md (hunk 4 above)
Comment (jairideout) [inline CHANGELOG.md]: That's probably why the diff looks a little odd

**Comment by RNAer** [comment]
comments addressed and one branch of travis passed. good to merge

**Pull request merged**